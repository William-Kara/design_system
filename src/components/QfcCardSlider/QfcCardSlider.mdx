import { ArgTypes, Meta } from '@storybook/blocks';
import { Story } from '@storybook/addon-docs';
import QfcCardSlider from '@/components/QfcCardSlider/QfcCardSlider.vue';

<Meta title="Modules/CardSlider" />

# Definition

Slider used to display a list of cards.

Slider will display as many cards as possible on the width available.

If there are more cards than can be displayed, the slider will be horizontally scrollable.

## Props

The following props are available for the QfcCardSlider component:

<ArgTypes of={QfcCardSlider} />

# Usage

To use QfcCardSlider, style needs to be imported separately.

```sass
@import '@website/design-system-vuejs/dist/styles/components/QfcCardSlider.scss';
```

```html
<template>
  <QfcCardSlider :cards="cards" />
</template>
```

# Exemple

<Story name="Primary" id="modules-cardslider--primary" />
