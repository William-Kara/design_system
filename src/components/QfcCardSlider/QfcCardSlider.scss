.qfc-slider {
  display: flex;
  gap: $q-generic-space-3;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
  scroll-snap-type: x mandatory;
  scrollbar-width: none;
  -ms-overflow-style: none;

  &__slide {
    flex-shrink: 0;
    scroll-snap-align: center;
    scroll-behavior: smooth;
  }
  &::-webkit-scrollbar {
    width: 0;
    background: transparent;
  }
}
