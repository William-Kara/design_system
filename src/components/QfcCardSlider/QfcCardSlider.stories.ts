import CardSlider from './QfcCardSlider.vue';
import type { Meta, StoryObj } from '@storybook/vue3';
import videoFile from '../../public/timer-animation.mp4';

const meta: Meta<typeof CardSlider> = {
  component: CardSlider,
  title: 'Modules/CardSlider',
  argTypes: {
    cards: {
      defaultValue: [],
    },
  },
};
export default meta;

type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    cards: [
      {
        title: 'Une plateforme accessible à tout moment',
        srcVideo: videoFile,
        icon: 'timer',
        size: 'medium',
        content:
          'Grâce à une plateforme accessible partout et à tout moment, vous pourrez retrouver vos formations en cours rapidement et reprendre là où vous en étiez.',
      },
      {
        title: 'Des parcours adaptés à vos besoins',
        icon: 'explorer',
        size: 'medium',
        content:
          'Nous vous offrirons une expérience d’apprentissage sur-mesure avec des parcours de formation adaptés à vos besoins. Chaque module sera pensé de telle manière à ce que vous ayez uniquement accès à l’essentiel.',
      },
      {
        title: 'De la théorie et surtout de la mise en pratique',
        icon: 'explorer',
        size: 'medium',
        content:
          'Parce que la théorie ça se pratique, nous avons décider de vous proposer de nombreux exercices par module et ainsi de vous faire gagner en confiance. ',
      },
    ],
  },
};
