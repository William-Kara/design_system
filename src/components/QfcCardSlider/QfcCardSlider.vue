<template>
  <div class="qfc-slider">
    <QfcCard
      v-for="({ icon, title, content, size, srcVideo }, index) in cards"
      :key="`card-container-${index}`"
      class="qfc-slider__slide"
      :icon="icon"
      :srcVideo="srcVideo"
      :size="size"
      :title="title"
      :content="content"
    />
  </div>
</template>

<script lang="ts" setup>
import QfcCard from '@/components/QfcCard/QfcCard.vue';

type Card = {
  srcVideo: string;
  icon: string;
  title: string;
  content: string;
  size: string;
};
interface Props {
  cards: Card[];
}
defineProps<Props>();
</script>

<style lang="scss" scoped src="./QfcCardSlider.scss" />
