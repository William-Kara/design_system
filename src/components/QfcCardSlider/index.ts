import QfcCardSlider from './QfcCardSlider.vue';

export { QfcCardSlider };
export default QfcCardSlider;
