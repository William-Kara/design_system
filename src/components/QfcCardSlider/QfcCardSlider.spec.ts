import { it, describe, beforeEach, expect } from 'vitest';

import { shallowMount } from '@vue/test-utils';
import QfcCardSlider from './QfcCardSlider.vue';

let wrapper: any = null;

const propsValues = {
  cards: [
    {
      title: 'Une plateforme accessible à tout moment',
      icon: 'timer',
      size: 'medium',
      content:
        'Grâce à une plateforme accessible partout et à tout moment, vous pourrez retrouver vos formations en cours rapidement et reprendre là où vous en étiez.',
    },
    {
      title: 'Des parcours adaptés à vos besoins',
      icon: 'explorer',
      size: 'medium',
      content:
        'Nous vous offrirons une expérience d’apprentissage sur-mesure avec des parcours de formation adaptés à vos besoins. Chaque module sera pensé de telle manière à ce que vous ayez uniquement accès à l’essentiel.',
    },
    {
      title: 'De la théorie et surtout de la mise en pratique',
      icon: 'explorer',
      size: 'medium',
      content:
        'Parce que la théorie ça se pratique, nous avons décider de vous proposer de nombreux exercices par module et ainsi de vous faire gagner en confiance. ',
    },
  ],
};

beforeEach(() => {
  wrapper = shallowMount(QfcCardSlider, {
    propsData: propsValues,
  });
});

describe('QfcCardSlider Component', () => {
  it('should mount properly', () => {
    expect(wrapper).toBeTruthy();
  });

  it('should have correct props', () => {
    expect(wrapper.props()).toEqual(propsValues);
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
