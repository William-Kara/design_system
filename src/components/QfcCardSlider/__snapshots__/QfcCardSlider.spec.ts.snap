// Vitest Snapshot v1

exports[`QfcCardSlider Component > should render correctly 1`] = `
"<div data-v-2eeab328=\\"\\" class=\\"qfc-slider\\">
  <qfc-card-stub data-v-2eeab328=\\"\\" icon=\\"timer\\" title=\\"Une plateforme accessible à tout moment\\" content=\\"Grâce à une plateforme accessible partout et à tout moment, vous pourrez retrouver vos formations en cours rapidement et reprendre là où vous en étiez.\\" size=\\"medium\\" class=\\"qfc-slider__slide\\"></qfc-card-stub>
  <qfc-card-stub data-v-2eeab328=\\"\\" icon=\\"explorer\\" title=\\"Des parcours adaptés à vos besoins\\" content=\\"Nous vous offrirons une expérience d’apprentissage sur-mesure avec des parcours de formation adaptés à vos besoins. Chaque module sera pensé de telle manière à ce que vous ayez uniquement accès à l’essentiel.\\" size=\\"medium\\" class=\\"qfc-slider__slide\\"></qfc-card-stub>
  <qfc-card-stub data-v-2eeab328=\\"\\" icon=\\"explorer\\" title=\\"De la théorie et surtout de la mise en pratique\\" content=\\"Parce que la théorie ça se pratique, nous avons décider de vous proposer de nombreux exercices par module et ainsi de vous faire gagner en confiance. \\" size=\\"medium\\" class=\\"qfc-slider__slide\\"></qfc-card-stub>
</div>"
`;
